subsys/bluetooth/host/l2cap.c

```diff
@@ -961,7 +961,8 @@ static void l2cap_chan_tx_init(struct bt_l2cap_le_chan *chan)
 	LOG_DBG("chan %p", chan);
 
 	(void)memset(&chan->tx, 0, sizeof(chan->tx));
-	atomic_set(&chan->tx.credits, 0);
+	/*In order to improve the packet sending speed of the sender, increase the initial credits value of tx to 30.*/
+	atomic_set(&chan->tx.credits, 30);
```

Comment:
this is not legal. you can't just set an arbitrary number of credits on the peer's endpoint.

Reply:
hi jori, do you mean that tx and rx credits should be set the same? thanks!

Reply:
no I mean you should not touch anything in `chan->tx`, as those params are set by the remote side.

Reply:
and there's a test in the tree that uses it:
https://github.com/jori-nordic/zephyr/blob/9569e4410475fb62125e769a13222882902d8c75/tests/bsim/bluetooth/host/l2cap/credits_seg_recv/src/main.c#L423